test/modules/load-gen/PriorityMessageSender.cs

```diff
@@ -4,15 +4,21 @@ namespace LoadGen
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using System.Net;
+    using System.Text;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Client;
     using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
     using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
 
     class PriorityMessageSender : LoadGenSenderBase
     {
         readonly Random rng = new Random();
+        private bool isFinished;
+        private int resultsSent;
```

Comment:
In relayer, these values were thread safe. Do they need to be thread safe here too?

Reply:
I made them thread safe in Relayer because relayer uses a callback on receiving input to send messages. So the messages can be handled asynchronously. It also needs to keep track of previous results and handle duplicates.

PriorityMessageSender is all synchronous, so we don't need to add any thread safety.